### DISCUSSION TITLE
How to use snow_2 output with FickianDiffusion algorithm

### DISCUSSION THREAD

**Discussion by ryanvolpi** [PMEAL/OpenPNM#1916] [Q&A] [answered]
Hello! I'd like to use the FickianDiffusion  algorithm on a network I extract using snow2 but I can't seem to find an example which lays out the steps in a way that works in the current version of openpnm. I think my issue is related to getting the network geometry, but I am not sure. As an example of what I am trying, see below. Any direction would be greatly appreciated!

```
import porespy as ps
import openpnm as op

im = ps.generators.blobs(shape=[50, 50, 50], porosity=0.6)
snow_output = ps.networks.snow2(im.astype(int), boundary_width = [[0,0],[0,0],[3,0]])
#create empty dummy network
pn = op.network.GenericNetwork()
#update openpnm pore network
pn.update(snow_output.network)
#prj = pn.project

air = op.phases.GenericPhase(network=pn)
f = op.models.phases.diffusivity.fuller
air.add_model(propname='pore.diffusivity',
              model=f,
              MA=0.032, MB=0.028, vA=16.6, vB=17.9)

phys = op.physics.GenericPhysics(network=pn, phase=air)#, geometry=geo
f = op.models.physics.diffusive_conductance.ordinary_diffusion
phys.add_model(propname='throat.diffusive_conductance',
               model=f)

fd = op.algorithms.FickianDiffusion(network=pn, phase=air)
fd.set_value_BC(pores=pn.pores('boundary'), values=1)
fd.set_value_BC(pores=[p for p in pn.pores() if not p in pn.pores('boundary')], values=0)
fd.run()
#print(fd.rate(pores=pn.pores('left')))
```

> C:\Users\volpir\PycharmProjects\Computer Vision Playground\venv\lib\site-packages\numba\np\ufunc\parallel.py:365: NumbaWarning: The TBB threading layer requires TBB version 2019.5 or later i.e., TBB_INTERFACE_VERSION >= 11005. Found TBB_INTERFACE_VERSION = 11004. The TBB threading layer is disabled.
>   warnings.warn(problem)
> WARNING    : No Geometry provided, phys_01 will not be associated with any locations 
> SOURCE     : openpnm.physics.GenericPhysics.__init__ 
> TIME STAMP : 2021-05-19 16:34:37,814
>
> ERROR      : throat.diffusive_conductance was not run since the following property is
>              missing: 'pore.area' 
> SOURCE     : openpnm.core.ModelsMixin._regen 
> TIME STAMP : 2021-05-19 16:34:37,817

> KeyError                                  Traceback (most recent call last)
> <ipython-input-4-2644439baf67> in <module>
>      27 fd.set_value_BC(pores=pn.pores('boundary'), values=1)
>      28 fd.set_value_BC(pores=[p for p in pn.pores() if not p in pn.pores('boundary')], values=0)
> ---> 29 fd.run()
>      30 #print(fd.rate(pores=pn.pores('left')))
> 
> ~\PycharmProjects\Computer Vision Playground\venv\lib\site-packages\openpnm\algorithms\ReactiveTransport.py in run(self, x0)
>     152         self._validate_settings()
>     153         # Check if A and b are well-defined
> --> 154         self._validate_data_health()
>     155         quantity = self.settings['quantity']
>     156         logger.info('Running ReactiveTransport')
> 
> ~\PycharmProjects\Computer Vision Playground\venv\lib\site-packages\openpnm\algorithms\GenericTransport.py in _validate_data_health(self)
>     790 
>     791         # Short-circuit subsequent checks if data are healthy
> --> 792         if np.isfinite(self.A.data).all() and np.isfinite(self.b).all():
>     793             return True
>     794         # Validate network topology health
> 
> ~\PycharmProjects\Computer Vision Playground\venv\lib\site-packages\openpnm\algorithms\GenericTransport.py in _get_A(self)
>     487     def _get_A(self):
>     488         if self._A is None:
> --> 489             self._build_A()
>     490         return self._A
>     491 
> 
> ~\PycharmProjects\Computer Vision Playground\venv\lib\site-packages\openpnm\algorithms\GenericTransport.py in _build_A(self)
>     466             except KeyError:
>     467                 raise Exception('Phase has not been defined for algorithm')
> --> 468             g = phase[gvals]
>     469             am = network.create_adjacency_matrix(weights=g, fmt='coo')
>     470             self._pure_A = spgr.laplacian(am).astype(float)
> 
> ~\PycharmProjects\Computer Vision Playground\venv\lib\site-packages\openpnm\phases\GenericPhase.py in __getitem__(self, key)
>      97                                prop=not_el + '.' + prop,
>      98                                mode='mean')
> ---> 99         vals = super().__getitem__(key)
>     100         return vals
> 
> ~\PycharmProjects\Computer Vision Playground\venv\lib\site-packages\openpnm\core\Base.py in __getitem__(self, key)
>     275             vals = super().__getitem__(key)
>     276         else:
> --> 277             raise KeyError(key)
>     278         return vals
>     279 
> 
> KeyError: 'throat.diffusive_conductance'

**Comment by jgostick** [comment, marked as answer]
The 'dev' version of Porespy (which I assume you're using since you have used ``snow2``) is not quite compatible with OpenPNM.  We are in the midst converting PoreSpy to V2, and then OpenPNM to V3, but everything takes longer than it was supposed to.  For now, I suggest you use the 'conda' or 'pip' versions of both packages, which are compatible with each other.

**Reply by ryanvolpi** [reply]
Oh, of course, that makes sense! Thank you.
